### ISSUE TITLE
Incorrect syntax highlighting for commented AGS script in dialog

### ISSUE THREAD

**Issue by morganwillcock** [adventuregamestudio/ags#1783]
**Describe the bug**
Within the dialog script editor it looks like the ordering of syntax highlighting that is being applied to embedded AGS script is incorrect, leading to portions of the script being incorrectly highlighted.

**AGS Version**
3.5.1.19

**Screenshot**
![dialog_syntax](https://user-images.githubusercontent.com/14958747/190925169-7c634962-4032-4a48-9d18-b3b27a355698.png)

**To Reproduce**
Create a dialog script with the following content:
```
Roger: I'm looking for my key.
  //if (player.HasInventory(iKey)) {
  //    player.Say("Actually, I'd better go.");
  //    return RUN_DIALOG_STOP_DIALOG;
  //}
```
The commented out code will not be fully highlighted as being commented out. The highlighting of the comment opening `//` on the 2nd line will also be incorrect.

**Expected behavior**
Comment opening syntax and commented out code should both be highlighted as a comment.

**Comment by ivan-mogilko** [comment]
@morganwillcock I opened a PR #1792 for 3.6.0, could you please try it out to double check that it works?

If it does, we might also apply this to 3.5.1 branch, as the fix seem simple enough.

**Comment by ivan-mogilko** [comment]
Merged the fix into 3.6.0. I might also backport to 3.5.1, although the commit cannot be applied directly as it used older version of Scintilla.

**Comment by ivan-mogilko** [comment]
Fixed in release-3.5.1 as 23c79a0f4b738b45147f5b752a2b4b4ff2d5ff36, will be a part of Patch 15
